Separated the button label 'Approve' from index.jelly file to properties file

nuupur9 opened the pull request.

oleg-nenashev commented:
Seems the work is in progress

oleg-nenashev commented:
@nuupur9 This approach is not going to work. Why don't you just use System Property?

nuupur9 commented:
Sure. i'll make the changes and push the code.

On Tue, Jun 28, 2016 at 5:57 PM, Oleg Nenashev <notifications@github.com>
wrote:

> @nuupur9 <https://github.com/nuupur9> This approach is not going to work.
> Why don't you just use System Property?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/promoted-builds-plugin/pull/94#issuecomment-229034140>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AR3PfPxotcixqoFe61hnCX_Q1q9Ar_iIks5qQRMxgaJpZM4I_DtG>
> .
>

oleg-nenashev commented on src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java:
```diff
@@ -52,8 +55,31 @@
+    	try {
+    		//to use Custom label, add deamon property 'PROMOTION_BUILD_APPROVE_BUTTON_CUSTOM_LABEL' to "/etc/init.d/jenkins" file in value to JAVA_CMD
+    		//sample: JAVA_CMD="$JENKINS_JAVA_CMD $JENKINS_JAVA_OPTIONS -DJENKINS_HOME=$JENKINS_HOME -DPROMOTION_BUILD_APPROVE_BUTTON_CUSTOM_LABEL=Execute -jar $JENKINS_WAR"
+    		executeButtonLabel = System.getProperty("PROMOTION_BUILD_APPROVE_BUTTON_CUSTOM_LABEL");
```
In Jenkins a common practice is to include class name into the beginning of he property name. Just see examples in this plugin

oleg-nenashev commented on src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java:
```diff
@@ -52,8 +55,31 @@
+    		
+    	} catch (Exception e) {
+    		e.printStackTrace();
+    		executeButtonLabel = "Approve";
```
🐛 You converted a localizable string to a non-localizable one. See https://wiki.jenkins-ci.org/display/JENKINS/Internationalization

oleg-nenashev commented on src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java:
```diff
@@ -52,8 +55,31 @@
+    //Code to fetch Approve Promotion Button label from properties file
+    private String executeButtonLabel;
+    
+    public String getExecuteButtonLabel() {
```
Add Javadoc. Also should include <code>@since TODO</code> and a meaningful description.

oleg-nenashev replied:
Also makes sense to add Nonnull annotation

nuupur9 commented:
Hi @oleg-nenashev , 

Can you please review the latest pull request.

oleg-nenashev commented:
[My comment](https://github.com/jenkinsci/promoted-builds-plugin/pull/94#discussion_r68931604) regarding the property name has not been addressed yet. "In Jenkins a common practice is to include class name into the beginning of he property name. Just see examples in this plugin".

The PR also needs at least one functional test for the new functionality

oleg-nenashev commented on src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java:
```diff
@@ -52,8 +55,25 @@
+    public String getExecuteButtonLabel() {
+    	String label = System.getProperty("MANUAL_CONDITION_APPROVE_BUTTON_CUSTOM_LABEL");
+    	if(null == label) {
+    		label = "Approve";
```
Tabs vs spaces

oleg-nenashev replied:
Also: My comment regarding localizable strings has not been addressed

nuupur9 replied:
Regarding naming of the property, can the property be named as 'ManualCondition_ApproveButtonCustomLabel" ?

Regarding Localization of Strings, can you point me the place where you've used localization in your code, so that I can make the changes accordingly.

oleg-nenashev replied:
The property should be something like <code>ManualCondition.class.getName() + ".approveButtonLabel"</code>.

Regarding internationalization, see the document link I've provided in the previous comment

oleg-nenashev commented on src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java:
```diff
@@ -114,7 +134,7 @@ public PromotionBadge isMet(PromotionProcess promotionProcess, AbstractBuild<?,?
 
         for (ManualApproval approval : approvals) {
-            if (approval.name.equals(promotionProcess.getName()))
+        	if (approval.name.equals(promotionProcess.getName()))
```
tabs vs spaces

oleg-nenashev commented on src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java:
```diff
@@ -52,8 +55,25 @@
+	//sample: JAVA_CMD="$JENKINS_JAVA_CMD $JENKINS_JAVA_OPTIONS -DJENKINS_HOME=$JENKINS_HOME -DMANUAL_CONDITION_APPROVE_BUTTON_CUSTOM_LABEL=CUSTOM_VALUE -jar $JENKINS_WAR"
+    
+    @Nonnull
+    private String executeButtonLabel;
```
This field is not being used anywhere

oleg-nenashev commented on src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java:
```diff
@@ -311,6 +331,7 @@ public String getDisplayName() {
         public ManualCondition newInstance(StaplerRequest req, JSONObject formData) throws FormException {
             ManualCondition instance = new ManualCondition();
             instance.users = formData.getString("users");
+            instance.executeButtonLabel = instance.getExecuteButtonLabel();
```
Unused field

nuupur9 replied:
This field is referred in following file
src/main/resources/hudson/plugins/promoted_builds/conditions/ManualCondition/index.jelly

oleg-nenashev replied:
The field reference is not required. stapler is able to get it from the getter method

nuupur9 commented:
@oleg-nenashev I've addressed 2 of the comments.
1. Updated the Property Name
2. Removed unused variable. 
Ill complete the test case for this and push it.
Can we take up localization in the next iteration, as this change is a priority.

oleg-nenashev commented:
Missed last comment :(
Is the work still in progress?

oleg-nenashev commented:
@nuupur9 Is this change still a priority?

oleg-nenashev commented:
No response
